sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -21,10 +21,9 @@ import org.apache.spark.sql.AnalysisException
 import org.apache.spark.sql.catalyst.{FunctionIdentifier, InternalRow, TableIdentifier}
 import org.apache.spark.sql.catalyst.expressions._
 import org.apache.spark.sql.catalyst.expressions.codegen.{CodegenContext, ExprCode}
-import org.apache.spark.sql.catalyst.parser.ParserUtils
 import org.apache.spark.sql.catalyst.plans.logical.{LeafNode, LogicalPlan, UnaryNode}
-import org.apache.spark.sql.catalyst.util.quoteIdentifier
 import org.apache.spark.sql.connector.catalog.{Identifier, TableCatalog}
+import org.apache.spark.sql.connector.catalog.CatalogV2Implicits.quoteIfNeeded
```

Comment:
Since `CatalogV2Implicits.quoteIfNeeded` is so widely used now, shall we move it to `org.apache.spark.sql.catalyst.util` and replace the `quoteIdentifier` there?

Reply:
Moving `quoteIfNeeded` to `org.apache.spark.sql.catalyst.util` and replacing `quoteIdentifier` with it seems reasonable.
Should we replace all the occurrences of `quoteIdentifier` in this PR?

Reply:
Ah I thought we already did. If not, let's just move `quoteIfNeeded` to `util`. We can replace `quoteIdentifier` completely in a follow-up PR.

Reply:
OK, I've moved it.